components/script/dom/htmlinputelement.rs

```diff
@@ -111,10 +133,48 @@ struct InputTypeColorShadowTree {
     color_value: Dom<HTMLDivElement>,
 }
 
+// FIXME: These styles should be inside UA stylesheet, but it is not possible without internal pseudo element support.
+const TEXT_TREE_STYLE: &str = "
+#input-editor::selection, #input-placeholder::selection {
```

Comment:
placeholder should not be selectable and therefore should not have `::selection` style.

Reply:
Sure, thanks!